MULE-16301: Review variable scoping docs for different Scopes (#375)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/batch-processing-concept.adoc

As a good practice, you should configure some mechanism for reporting on failed or successful records to facilitate further action where required.
--

=== Variable Propagation

Every processed record of the Batch Job Instance will start with the same variables and values as before the execution of the block. Every record will have their own set of variables, so new variables or modifications of already existing ones while processing a record will not be visible while processing another one. For each record, those variables (and modifications) will be propagated though the different Batch Steps. For example, if record R1 sets a variable `varName: "hello"`, record R2 sets `varName: "world"` and record R3 does not sets this variable, then in the next step R1 will see the value `"hello"`, R2 the value `"world"` and R3 will not see any value for that variable.

> In the On Complete phase, non of this variables (and not even the original ones) are visible. Only the final result is available in this phase. Moreover, since the Batch Job Instance executes asynchronously from the rest of the flow, then any variable set neither in a Batch Step, nor in the On Complete phase will be visible outside the Batch Scope.

Reviewer: `non` -> `none`

Reviewer: `any variable` -> `no variable`?

Author: ```suggestion
In the On Complete phase, none of this variables (and not even the original ones) are visible. Only the final result is available in this phase. Moreover, since the Batch Job Instance executes asynchronously from the rest of the flow, then no variable set neither in a Batch Step, nor in the On Complete phase will be visible outside the Batch Scope.
```
